tests/foreman/cli/test_repository.py

```diff
@@ -911,3 +918,149 @@ def test_positive_upload_content(self):
             "Successfully uploaded file '{0}'".format(RPM_TO_UPLOAD),
             result[0]['message'],
         )
+
+
+class OstreeRepositoryTestCase(CLITestCase):
+    """Ostree Repository CLI tests."""
+
+    @classmethod
+    @skip_if_os('RHEL6')
+    def setUpClass(cls):
+        """Create an organization and product which can be re-used in tests."""
+        super(OstreeRepositoryTestCase, cls).setUpClass()
+        cls.org = make_org()
+        cls.product = make_product({u'organization-id': cls.org['id']})
+
+    def _make_repository(self, options=None):
+        """Makes a new repository and asserts its success"""
+        if options is None:
+            options = {}
+
+        if options.get('product-id') is None:
+            options[u'product-id'] = self.product['id']
+
+        return make_repository(options)
+
+    @tier1
+    def test_positive_create_ostree_repo(self):
+        """Create a ostree repository
+
+        @id: a93c52e1-b32e-4590-981b-636ae8b8314d
+
+        @Assert: ostree repository is created
+        """
+        for name in valid_data_list():
+            with self.subTest(name):
+                new_repo = self._make_repository({
+                    u'name': name,
+                    u'content-type': u'ostree',
+                    u'publish-via-http': u'false',
+                    u'url': FEDORA23_OSTREE_REPO,
+                })
+        self.assertEqual(new_repo['name'], name)
+        self.assertEqual(new_repo['content-type'], u'ostree')
+
+    @tier1
+    @skip_if_bug_open('bugzilla', 1370108)
+    def test_negative_create_ostree_repo_with_checksum(self):
+        """Create a ostree repository with checksum type
+
+        @id: a334e0f7-e1be-4add-bbf2-2fd9f0b982c4
+
+        @Assert: Validation error is raised
+        """
+        for checksum_type in u'sha1', u'sha256':
+            with self.subTest(checksum_type):
+                with self.assertRaisesRegex(
+                    CLIFactoryError,
+                    u'Validation failed: Checksum type cannot be set for '
+                    'non-yum repositories'
+                ):
```

Comment:
nitpick: as we now support python 2.7+, you may use following syntax:
```python
with A() as X, B() as Y, C() as Z:
    do_something()
```
to avoid nested context managers.